### ISSUE TITLE
Select on transformed port throws ClassCastException

### ISSUE THREAD

**Issue by gkristjansson** [puniverse/quasar#129]
It seems that its not possible to do a select on a receive port that has been transformed.

```
Channel<Message> ch = Channels.newChannel(-1)
ReceivePort rp = Channels.<Message, Message>map(ch, (m) -> m) 
....
SelectAction<Message> sa = Selector.select(timeout, rp, ...);
```

This will throw a ClassCastException.

Is there a know workaround ?

**Comment by pron** [comment]
Right, we should add such support (although it may not be possible for all transformations). As a workaround you can add a [fiber-transform](http://docs.paralleluniverse.co/quasar/javadoc/co/paralleluniverse/strands/channels/Channels.html#fiberTransform-co.paralleluniverse.strands.channels.ReceivePort-co.paralleluniverse.strands.channels.SendPort-co.paralleluniverse.strands.SuspendableAction2-) that reads from one channel, transforms, and writes to another, and then select on the fiber transform's output channel.
